quinn/src/connection.rs

```diff
@@ -651,6 +651,37 @@ impl Connection {
         // May need to send MAX_STREAMS to make progress
         conn.wake();
     }
+
+    /// Waits until the connection received TLS resumption tickets
+    ///
+    /// Yields `true` once resumption tickets were received. Resolves immediately
+    /// if tickets were already received, otherwise it resolves once tickets arrive.
+    /// If the server does not send any tickets, the returned future will remain pending forever.
+    ///
+    /// This should only be used on the client side. On the server side, it will
+    /// always resolve immediately and yield `false`.
+    pub fn resumption_tickets_received(&self) -> impl Future<Output = bool> + Send + 'static {
```

Comment:
How is the user meant to actually get the ticket? If the expectation is that they'll still be going through rustls's `ClientSessionStore`, would it be simpler to bypass quinn entirely and use a `ClientSessionStore` that notifies you directly on insert?